src/AppInstallerCLICore/Workflows/ManifestComparator.h

```diff
@@ -26,16 +32,19 @@ namespace AppInstaller::CLI::Workflow
     };
 
     // Class in charge of comparing manifest entries
-    class ManifestComparator
+    struct ManifestComparator
     {
-    public:
-        ManifestComparator(const Execution::Args&) {}
+        ManifestComparator(const Execution::Args&, std::optional<std::map<std::string, std::string>> installationMetadata) : 

```

Comment:
>optional [](start = 56, length = 8)

Do you need an optional here, and empty map should work just fine as well, right?  I don't see how "map present, but key missing" is an different from "map missing".